test/distributed/pipelining/test_backward.py

```diff
@@ -117,7 +117,7 @@ def test_stage_backward_weight(self):
         )
 
         # backward of loss with respect to weights
-        dweights = stage_backward_weight(mod.parameters(), param_groups)
+        stage_backward_weight(mod.parameters(), param_groups, retain_graph=True)
```

Comment:
Why retain graph=true? Is it because we reuse the same graph for subsequent micro batches? Or does each micro batch have its own graph?

Reply:
Oh this is a test. Never mind